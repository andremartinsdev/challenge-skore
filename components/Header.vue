<!-- Please remove this file from your project -->
<template>
  <div>
    <header class="header-conteiner">
      <div>
        <span class="header-content fw-300">Conte√∫dos</span>
        <span class="header-content fw-600">LGPD</span>
      </div>
      <div>
        <img
          src="~/assets/images/lgpd.png"
          class="header-imagen"
          alt=""
          srcset=""
        />
      </div>
    </header>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'TutorialTeste',
})
</script>

<style lang="scss" scoped>
header {
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #1e3c72;

  .header-content {
    font-size: 45px;
    margin: 10px;
  }

  img {
    width: 150px;
  }
}

@include for-phone-only {
  header {
    width: 100%;
    img {
      width: 100px;
    }
    .header-content {
      font-size: 25px;
      margin: 5px;
    }
  }
}
</style>
