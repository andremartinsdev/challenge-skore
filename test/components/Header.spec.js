import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'

describe('Header.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Header)
  })

  it('should render Header', () => {
    expect(wrapper.find('.header-conteiner').isVisible()).toBe(true)
  })

  it('should render Header', () => {
    expect(wrapper.text()).toContain('ConteÃºdos')
    expect(wrapper.text()).toContain('LGPD')
  })

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
