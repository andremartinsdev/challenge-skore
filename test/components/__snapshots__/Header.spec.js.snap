// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders properly 1`] = `
"<div>
  <header class=\\"header-conteiner\\">
    <div><span class=\\"header-content fw-300\\">Conte√∫dos</span> <span class=\\"header-content fw-600\\">LGPD</span></div>
    <div><img src=\\"~/assets/images/lgpd.png\\" alt=\\"\\" srcset=\\"\\" class=\\"header-imagen\\"></div>
  </header>
</div>"
`;
